packages/go_router/lib/src/information_provider.dart

```diff
@@ -79,18 +80,35 @@ class GoRouteInformationProvider extends RouteInformationProvider
     required String initialLocation,
     required Object? initialExtra,
     Listenable? refreshListenable,
+    bool routerNeglect = false,
   })  : _refreshListenable = refreshListenable,
         _value = RouteInformation(
           uri: Uri.parse(initialLocation),
           state: RouteInformationState<void>(
               extra: initialExtra, type: NavigatingType.go),
         ),
-        _valueInEngine = _kEmptyRouteInformation {
+        _valueInEngine = _kEmptyRouteInformation,
+        _routerNeglect = routerNeglect {
     _refreshListenable?.addListener(notifyListeners);
   }
 
+  /// Notifies the platform for a route information change.
+  @visibleForTesting
+  @internal
+  Future<void> systemNavigatorRouteInformationUpdated(
```

Comment:
This is not necessary, you can listen to platform message during test using
See example here https://github.com/flutter/flutter/blob/d39550fb528c863431f1e3b754d4279defd82081/packages/flutter/test/services/system_navigator_test.dart#L34

Reply:
Ah, great, will update the code.